src/awesome/class.rs

```diff
@@ -0,0 +1,163 @@
+//! Utility methods and constructors for Lua classes
+
+use std::default::Default;
+use std::rc::Rc;
+use rlua::{self, Lua, ToLua, Table, UserData, AnyUserData, Value, Function};
+use super::object::Object;
+use super::property::Property;
+
+pub type Allocator = Rc<Fn(&Lua) -> rlua::Result<Object>>;
+pub type Collector = Rc<Fn(Object)>;
+pub type PropF<'lua> = rlua::Function<'lua>;
+pub type Checker = Rc<Fn(Object) -> bool>;
+
+#[derive(Debug)]
+pub struct Class<'lua> {
+    table: Table<'lua>
+}
+
+#[derive(Clone)]
+pub struct ClassState {
```

Comment:
Need comment explaining where the missing fields are compared to the C version (they are stored in Lua). Add a getter for them